File: code/osgi/base/com.apollo.demos.osgi.base.api.scala/src/main/scala/com/apollo/demos/osgi/base/api/scala/LogHelper.scala
package com.apollo.demos.osgi.base.api.scala

import org.slf4j.Logger

trait LogHelper {
  def trace(logger: Logger, msg: String, arguments: Any*)
  def debug(logger: Logger, msg: String, arguments: Any*)
  def info(logger: Logger, msg: String, arguments: Any*)
  def warn(logger: Logger, msg: String, arguments: Any*)
  def error(logger: Logger, msg: String, arguments: Any*)
}

File: code/base/jsr269/com.apollo.demos.base.jsr269/src/main/java/com/apollo/demos/base/jsr269/LogHelper.java
/*
 * 此代码创建于 2022年3月16日 下午10:05:22。
 */
package com.apollo.demos.base.jsr269;

public interface LogHelper {

    static void log(Runnable task, String level) {
        System.out.println(level + ":start");
        task.run();
        System.out.println(level + ":end");
    }

}
